### PR TITLE
Clarify use of type-specific facets in a union type

### LINKED ISSUE #711: Clarify use of type-specific facets in a union type
The [description of the union type](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#union-type) makes no mention of how to apply facets and the restrictions thereof.

### Proposed clarification
- Add that `a union type MAY have any of its super types facets as long as all super types participating in that union accept those facets`

#### Examples
```
types:
  Foo: number
  Bar: integer
  FooBar:
    type: Foo | Bar
    minimum: 1 # valid because both "Foo" (number) and "Bar" (integer) all accept "minimum"
```
```
types:
  Foo: number
  Bar: integer
  Qux: string
  FooBarQux:
    type: Foo | Bar | Qux
    minimum: 1 # invalid because "Qux" (string) does not accept the "minimum" facet
```
```
types:
  Foo: number
  Bar: integer
  Qux:
    type: string
    facets:
      minimum: number
  FooBarQux:
    type: Foo | Bar | Qux
    minimum: 1 # valid because "Qux" (string) has a user-defined facet "minimum"
```

### PR DESCRIPTION
**Pull request by jstoiko**
Closes #711

### CODE REVIEW

**Comment by BlockLucas** [comment]
Hi!

Unions didn’t have facets in 1.0.
For us is not a PATCH change as an API with the following cases (defined in this PR), would be invalid in RAML 1.0:

```yaml
types:
  Foo: number
  Bar: integer
  Qux:
    type: string
    facets:
      minimum: number
  FooBarQux:
    type: Foo | Bar | Qux
    minimum: 1 # valid because "Qux" (string) has a user-defined facet "minimum"
```

```yaml
types:
  Foo: number
  Bar: integer
  FooBar:
    type: Foo | Bar
    minimum: 1 # valid because both "Foo" (number) and "Bar" (integer) all accept "minimum"
```

**Comment by jstoiko** [comment]
Where do you see that the union type cannot have [facets](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md)?

File: versions/raml-10/raml-10.md [hunk 1]
```diff
@@ -1170,6 +1170,42 @@ types:
    HomeAnimal: [ HasHome | IsOnFarm ,  Dog | Cat | Parrot ]
 ```
 
+A union type MAY use facets defined by any of its member types as long as all member types in the union accept those facets.
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1173]:
```suggestion
A `union` type may use facets defined by any of its member types, as long as all member types in the union accept those facets, for example:
```

File: versions/raml-10/raml-10.md [hunk 2]
```diff
@@ -1170,6 +1170,42 @@ types:
...
+
+For example:
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1175]:
```suggestion
For example: (delete)
```

File: versions/raml-10/raml-10.md [hunk 3]
```diff
@@ -1170,6 +1170,42 @@ types:
...
+
+```yaml
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1177]: Add the whole example in a code tag, like this:
[source, text, linenums]
----
Place the whole example here.
----

File: versions/raml-10/raml-10.md [hunk 4]
```diff
@@ -1170,6 +1170,42 @@ types:
...
+    facets:
+      minimum: number
+  FooBarQux:
+    type: Foo | Bar | Qux
+    minimum: 1 # valid because "Qux" (string) has a user-defined facet "minimum"
+```
+
 #### Using XML and JSON Schema
 
 RAML allows the use of XML and JSON schemas to describe the body of an API request or response by integrating the schemas into its data type system.
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:1211]:
```suggestion
RAML allows you to use XML and JSON schemas to describe the body of an API request or response by integrating the schemas into its data type system.
```

**Review by priyaayyar — CHANGES_REQUESTED** [review]

File: versions/raml-10/raml-10.md (hunk 4 above)
Comment (dukesphere) [reply versions/raml-10/raml-10.md:1211]: I'd decline the suggested change and keep the original as-is. This and a number of other changes are applying our doc style guidelines but would be inconsistent with the style of the existing doc. So I think our copyedits should be minimal.

File: versions/raml-10/raml-10.md (hunk 1 above)
Comment (dukesphere) [inline versions/raml-10/raml-10.md:1173]:
```suggestion
A union type MAY use facets defined by any of its member types as long as all member types in the union accept those facets, for example:
```

File: versions/raml-10/raml-10.md (hunk 2 above)
Comment (dukesphere) [inline versions/raml-10/raml-10.md:1175]:
```suggestion
```

Comment (dukesphere) [reply versions/raml-10/raml-10.md:1175]: use suggested change below, decline above

File: versions/raml-10/raml-10.md (hunk 3 above)
Comment (dukesphere) [reply versions/raml-10/raml-10.md:1177]: Priya, this Markdown,  not asciidoc.

Comment (dukesphere) [reply versions/raml-10/raml-10.md:1177]: @jstoiko Please disregard the suggestion above.

File: versions/raml-10/raml-10.md (hunk 1 above)
Comment (priyaayyar) [reply versions/raml-10/raml-10.md]: Tring to make it less wordy. So here is one more attempt:
A union type MAY use facets defined by any of its member types, if the member types in the union accept those facets, for example:

**Review by dukesphere — APPROVED** [review]

**Review by priyaayyar — APPROVED** [review]

**Pull request merged**